submission/programs/adlbc.R

```diff
@@ -0,0 +1,227 @@
+
+# create labels
+metacore <- spec_to_metacore("adam/TDF_ADaM - Pilot 3 Team updated.xlsx", where_sep_sheet = FALSE)
+
+adlbc_spec <- metacore %>%
+  select_dataset("ADLBC")
+
+# Formats -----------------------------------------------------------------
+## map parameter code and parameter
+format_paramn <- function(x) {
+  case_when(
+    x == "SODIUM" ~ 18,
+    x == "K" ~ 19,
+    x == "CL" ~ 20,
+    x == "BILI" ~ 21,
+    x == "ALP" ~ 22,
+    x == "GGT" ~ 23,
+    x == "ALT" ~ 24,
+    x == "AST" ~ 25,
+    x == "BUN" ~ 26,
+    x == "CREAT" ~ 27,
+    x == "URATE" ~ 28,
+    x == "PHOS" ~ 29,
+    x == "CA" ~ 30,
+    x == "GLUC" ~ 31,
+    x == "PROT" ~ 32,
+    x == "ALB" ~ 33,
+    x == "CHOL" ~ 34,
+    x == "CK" ~ 35
+  )
+}
+# Add supplemental information --------------------------------------------
+sup <- supplb %>%
+  select(STUDYID, USUBJID, IDVAR, IDVARVAL, QNAM, QLABEL, QVAL) %>%
+  pivot_wider(
+    id_cols = c(STUDYID, USUBJID, IDVARVAL),
+    names_from = QNAM,
+    values_from = QVAL
+  ) %>%
+  mutate(LBSEQ = as.numeric(IDVARVAL)) %>%
+  select(-IDVARVAL)
+
+adlb00 <- lb %>%
+  left_join(sup, by = c("STUDYID", "USUBJID", "LBSEQ")) %>%
+  filter(LBCAT == "CHEMISTRY")
+
+# ADSL information --------------------------------------------------------
+
+adsl <- adsl %>%
+  select(
+    STUDYID, SUBJID, USUBJID, TRT01PN, TRT01P, TRT01AN, TRT01A, TRTSDT, TRTEDT, AGE, AGEGR1, AGEGR1N, RACE, RACEN, SEX,
+    COMP24FL, DSRAEFL, SAFFL
+  )
+
+
+
+adlb01 <- adlb00 %>%
+  left_join(adsl, by = c("STUDYID", "USUBJID"))
+
+# Dates -------------------------------------------------------------------
+
+adlb02 <- adlb01 %>%
+  derive_vars_dtm(
+    new_vars_prefix = "A",
+    dtc = LBDTC,
+    highest_imputation = "s",
+    ignore_seconds_flag = T
+  ) %>%
+  derive_vars_dt(
+    new_vars_prefix = "A",
+    dtc = LBDTC,
+    highest_imputation = "n"
+  ) %>%
+  derive_vars_dy(reference_date = TRTSDT, source_vars = vars(ADT))
+
+
+
+# AVAL(C) -----------------------------------------------------------------
+# No imputations are done for values below LL or above UL
+
+adlb03 <- adlb02 %>%
+  mutate(
+    AVAL = LBSTRESN,
+    AVALC = ifelse(!is.na(AVAL), LBSTRESC, NA)
+  )
+
+# Parameter ---------------------------------------------------------------
+
+adlb04 <- adlb03 %>%
+  mutate(
+    PARAM = paste0(LBTEST, " (", LBSTRESU, ")"),
+    PARAMCD = LBTESTCD,
+    PARAMN = format_paramn(LBTESTCD),
+    PARCAT1 = "CHEM" # changed to match prod dataset
+  )
+
+# Baseline ----------------------------------------------------------------
+## updating to use admiral programming
+
+
+adlb05 <- adlb04 %>%
+  mutate(ABLFL = LBBLFL) %>%
+  derive_var_base(
+    by_vars = vars(STUDYID, USUBJID, PARAMCD),
+    source_var = AVAL,
+    new_var = BASE
+  ) %>%
+  derive_var_chg()
+
+
+# VISITS ------------------------------------------------------------------
+
+eot <- adlb05 %>%
```

Comment:
We also need to include subjects that did not do their end of treatment visit for lab 'Week 24" eg subjects 1 01-716-1103
2 01-716-1441
For these we need to select last visit before week 24

Can we try:

eot <- adlb05 %>%
  arrange(STUDYID, USUBJID, PARAMCD, AVISITN) %>%
  filter(VISITNUM <= 12) %>%
  slice(n()) %>%
  mutate(    AVISIT = "End of Treatment",
    AVISITN = 99
)

Reply:
Hi Steven!

This didn't solve our problem. Some of the observations are from week 24 (so subjects made it through entire trial) and some are from before week 24. 

I pushed some code to generate a table containing the missing observations. If we can't determine the best way to identify these records as EOT then maybe we document this finding?

Reply:
@SHAESEN2 can we resolve this now?

Reply:
As the QC against the original dataset is passing now, I will go ahead and resolve this for the time being. @SHAESEN2 please follow up if any further questions.